### PR TITLE
Handle case when `node.receiver` is `nil` in `Style/InvertibleUnlessCondition`

### PR DESCRIPTION
**Pull request by shioyama**
`Style/InvertibleUnlessCondition` changes in https://github.com/rubocop/rubocop/pull/12562 caused this cop to fail in the case where the node `receiver` is `nil`. See the failing test. I made a fix.

* [X] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by shioyama** [comment]
To complexity metrics are failing. I fundamentally disagree with these so if you'd like to fix them, please edit the changes:

```
Offenses:

lib/rubocop/cop/style/invertible_unless_condition.rb:102:9: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for preferred_send_condition is too high. [8/7]
        def preferred_send_condition(node) ...
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/rubocop/cop/style/invertible_unless_condition.rb:102:9: C: Metrics/PerceivedComplexity: Perceived complexity for preferred_send_condition is too high. [9/8]
        def preferred_send_condition(node) ...
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: lib/rubocop/cop/style/invertible_unless_condition.rb [hunk 1]
```diff
@@ -100,22 +100,24 @@ def preferred_condition(node)
...
-          end
 
-          if node.parenthesized?
-            return "#{receiver_source}.#{inverse_method_name}(#{argument_list})"
+          if node.operator_method?
+            "#{receiver_source} #{inverse_method_name} #{argument_list}"
+          elsif node.parenthesized?
+            "#{inverse_method_source}(#{argument_list})"
+          else
+            "#{inverse_method_source} #{argument_list}"
```
Comment (shioyama) [inline lib/rubocop/cop/style/invertible_unless_condition.rb:119]: To me a series of `if`/`else` is much clearer than a bunch of guard clauses.

**Comment by shioyama** [comment]
cc @sambostock

**Comment by sambostock** [comment]
🤦 I missed your ping @shioyama and ended up coincidentally opening #12711 to fix the same bug. I'll leave it open for comparison though, as the approach is _slightly_ different, and it has a couple more tests.

**Comment by shioyama** [comment]
Thanks! I'll close this, no worries.

**Pull request closed without merging**